core/src/Cuda/Kokkos_Cuda_Parallel_Team.hpp

```diff
@@ -106,7 +106,12 @@ class TeamPolicyInternal<Kokkos::Cuda, Properties...>
             (size_t)impl_vector_length(),
             (size_t)team_scratch_size(0) + 2 * sizeof(double),
             (size_t)thread_scratch_size(0) + sizeof(double));
-    return block_size / impl_vector_length();
+    int max_block_size = block_size / impl_vector_length();
+    if (max_block_size <= 0)
+      Kokkos::Impl::throw_runtime_exception(std::string(
```

Comment:
Isn't Damien trying to get rid of all exceptions?

Reply:
https://github.com/kokkos/kokkos/pull/7386/files#diff-96b736d454cbbe8522519e7a007cb1b70529f444131e0d02dfee3ff77c5b8d2d aborts in `Kokkos::Impl::throw_runtime_exception` if exceptions are disabled so this is not conflicting.

I choose to throw an exception here because that is what we are doing in similar cases. If users should be able to catch these exceptions (e.g., to try again with less local scratch memory), we should probably use a distinguishable exception type here but that seems out-of-scope for this pull request.